runtime/gc_modron_startup/mmparse.cpp

```diff
@@ -1209,21 +1209,54 @@ gcParseSovereignArguments(J9JavaVM *vm)
 		}
 	}
 
-	if (-1 != FIND_ARG_IN_VMARGS(EXACT_MATCH, "-XX:+HeapManagementMXBeanCompatibility", NULL)) {
-
-		extensions->_HeapManagementMXBeanBackCompatibilityEnabled = true;
-	}
-	
-	if (-1 != FIND_ARG_IN_VMARGS(EXACT_MATCH, "-XX:-HeapManagementMXBeanCompatibility", NULL)) {
-		extensions->_HeapManagementMXBeanBackCompatibilityEnabled = false;
-	}
 	return 1;
 
 _error:
 	return 0;
 
 }
 
+static UDATA
+gcParseXXArguments(J9JavaVM *vm)
+{
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+	MM_GCExtensions *extensions = MM_GCExtensions::getExtensions(vm);
+
+	{
+		IDATA heapManagementMXBeanCompatibilityIndex = -1;
+		IDATA noHheapManagementMXBeanCompatibilityIndex = -1;
+
+		if (-1 != FIND_ARG_IN_VMARGS(EXACT_MATCH, "-XX:+HeapManagementMXBeanCompatibility", NULL)) {
+
+			extensions->_HeapManagementMXBeanBackCompatibilityEnabled = true;
+		}
+
+		if (-1 != FIND_ARG_IN_VMARGS(EXACT_MATCH, "-XX:-HeapManagementMXBeanCompatibility", NULL)) {
+			if (noHheapManagementMXBeanCompatibilityIndex > heapManagementMXBeanCompatibilityIndex) {
+				extensions->_HeapManagementMXBeanBackCompatibilityEnabled = false;
+			}
+		}
+	}
+
+	{
+		IDATA useGCStartupHintsIndex = -1;
+		IDATA noUseGCStartupHintsIndex = -1;
```

Comment:
The pattern we use in the VM code is something like:
```c

IDATA useGCStartupHintsIndex = FIND_ARG_IN_VMARGS(EXACT_MATCH, "-XX:+UseGCStartupHints", NULL);
IDATA noUseGCStartupHintsIndex = FIND_ARG_IN_VMARGS(EXACT_MATCH, "-XX:-UseGCStartupHints", NULL);
if (useGCStartupHintsIndex >= noUseGCStartupHintsIndex) {
    extensions->useGCStartupHints = true;
} else {
    extensions->useGCStartupHints = false;
}
```

which removes the default `-1` values and assignments from the if statements

Reply:
This looks simpler, but has a problem: it always sets one value, so this logic must be aware of the default value previously set (in our case, at completely different part of the code), and must be careful to not overwrite it, if none of the two options are specified. Hence, it's prone to errors in case of other people cloning the code.

I'll try to find some compromise.